buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectTraverser.java

```diff
@@ -141,21 +156,47 @@ private void traverse(
     String nextDirectory = directories.peek();
     for (File f : files) {
       String fileName = f.getName();
-      if (f.isFile() && fileName.endsWith(".ftl")) {
+      if (f.isFile()) {
+        String fileText = "";
         Path absoluteFilePath = f.toPath();
+        try {
+          fileText = new String(Files.readAllBytes(absoluteFilePath),
+              StandardCharsets.UTF_8);
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
         Path rootPath = rootDirectory.toPath();
         String filePath = absoluteFilePath.subpath(rootPath.getNameCount() - 1,
             absoluteFilePath.getNameCount()).toString();
         // For windows paths change separator to forward slash.
         filePath = filePath.replace('\\', '/');
-        try {
-          String fileText = new String(Files.readAllBytes(absoluteFilePath),
-              StandardCharsets.UTF_8);
-          dialectGenerate.processFile(fileText, extractedData, filePath);
-        } catch (IOException e) {
-          e.printStackTrace();
-        } catch (IllegalStateException e) {
-          e.printStackTrace();
+        if (fileName.endsWith(".ftl")) {
+          try {
+            dialectGenerate.processFile(fileText, extractedData, filePath);
+          } catch (IllegalStateException e) {
+            e.printStackTrace();
+          }
+        } else if (fileName.equals("keywords.yaml")) {
+          // This is required to get past the license comment. The part after
+          // follows a JSON format so it can be parsed as such.
+          fileText = fileText.substring(fileText.indexOf("{"));
+          JSONObject json = (JSONObject) new JSONTokener(fileText).nextValue();
+          JSONArray keywordsJson = json.isNull("keywords")
+            ? null
+            : json.getJSONArray("keywords");
+          JSONArray nonReservedKeywordsJson = json.isNull("nonReservedKeywords")
```

Comment:
some inconsistencies with using Keywords vs KeyWords in this change eg in DialectGenerate.java it uses KeyWords but here it's Keywords

Reply:
My 2 cents: this should be ("keywords"/"Keywords") rather than ("keyWords"/"KeyWords") everywhere. This aligns with the pattern of only uppercasing letters that demarcate a new/distinct work in a variable name. However, "keyword" is just a single word.

Reply:
I agree under normal circumstances but NonReservedKeyWords() (the function) is referenced deeply in Calcite and created some nasty build errors when I tried to change it. It also wasn't obvious where changes needed to be made.

Reply:
Does it resolve the issues if you update the reference here (as well as the calls to the function in the parserImpls files)?

https://github.com/googleinterns/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/parser/SqlAbstractParserImpl.java#L587

Reply:
I'm not able to find any other references to this function name besides the one linked above, the one in the root parserImpls.ftl IdentifierSegment(), and one in Dialect1's SqlHostVariable().

Are there other references besides these?

Reply:
Oh, that may work. I'll try it and report back.